File: javascript/util/pandigital.js
function isPandigital(num, excludeZero) {
  let digits;

  if (typeof num === `number`) {
    // if a number was passed as the argument
    digits = new Map();

    while (num > 0) {
      const mod = num % 10;
      if ((excludeZero && mod === 0) || digits.has(mod)) {
        return false;
      }

      digits.set(mod, 1);
      num = Math.floor(num / 10);
    }

    return true;
  }
  if (typeof num === `string`) {
    // if a string was passed as the argument
    digits = num.split(``);
  } else if (Array.isArray(num)) {
    // if an array was passed as the argument
    digits = num;
  }

  // sort ("clone" the array in case an array was passed by reference)
  const digitsSorted = [...digits].sort();

  if (excludeZero && digitsSorted[0] === `0`) {
    return false;
  }

  for (let i = 1; i < digitsSorted.length; i++) {
    if (digitsSorted[i] === digitsSorted[i - 1]) {
      return false;
    }
  }

  return true;
}

module.exports = isPandigital;

File: python/util/pandigital.py
import math


def is_pandigital(num, exclude_zero):
  if type(num) == int:
    # if a number was passed as the argument
    digits = {}

    while num > 0:
      mod = num % 10
      if (exclude_zero == True and mod == 0) or digits.get(mod) is not None:
        return False

      digits[mod] = 1
      num = math.floor(num / 10)

    return True
  elif type(num) == str:
    # if a string was passed as the argument
    digits = list(num)
    digits.sort()

    if exclude_zero == True and digits[0] == '0':
      return False
  elif type(num) == list:
    # if a list was passed as the argument
    digits = num[:]
    digits.sort()

    if exclude_zero == True and digits[0] == 0:
      return False

  for i in range(1, len(digits)):
    if digits[i] == digits[i - 1]:
      return False

  return True
